docs/Lua API.rst

```diff
@@ -2984,6 +2984,66 @@ Predefined instance methods:
 
 To avoid confusion, these methods cannot be redefined.
 
+customRawData
+=============
+
+A utility allowing for custom tags to be added to the raw definitions for modding purposes:
+
+* ``customRawData.getTag(typeDefinition, tag[, ...])``
+
+  Where ``typeDefinition`` is a raw struct as seen in ``df.global.world.raws``, e.g:
+  ``dfhack.gui.getSelectedItem().subtype``, ``tag`` is the name of the custom tag you
+  want to get, and the remaining arguments correspond to the arguments in the tag as
+  booleans denoting whether the argument in the tag is a string or a number. ``true``
+  for string, false (or absent) for number. It will then be returned using single or
+  multiple return values. If the tag is not present, the result is false, if it is
+  present but has no arguments, the result is true.
```

Comment:
will the caller always know whether to expect strings or numbers for a particular tag? can we always return numbers if they are numbers and let the caller convert them to strings if desired? alternately, can we just always return strings and let the caller convert the numbers if they know to expect numbers? either way it would solve the issue of the cache returning unexpected types.

Reply:
Hmm... sure, okay. I'll remove the bool business.